Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -1309,9 +1321,50 @@ void NetworkStorageManager::setStorageSiteValidationEnabled(bool enabled)
     });
 }
 
+void NetworkStorageManager::addAllowedSitesForConnectionInternal(IPC::Connection::UniqueID connection, const Vector<WebCore::RegistrableDomain>& sites)
+{
+    assertIsCurrent(workQueue());
+
+    if (!m_allowedSitesForConnections)
+        return;
+
+    auto& allowedSites = m_allowedSitesForConnections->ensure(connection, []() {
```

Comment:
You can just use `add()`, constructing an empty HashSet is cheap (it's essentially a null pointer).